api/elections.go

```diff
@@ -250,6 +261,87 @@ func (a *API) electionVotesHandler(msg *apirest.APIdata, ctx *httprouter.HTTPCon
 	return ctx.Send(data, apirest.HTTPstatusCodeOK)
 }
 
+// GET /elections/<electionID>/scrutiny
+// returns the consensus results of an election
+func (a *API) electionScrutinyHandler(msg *apirest.APIdata, ctx *httprouter.HTTPContext) error {
+	electionID, err := hex.DecodeString(util.TrimHex(ctx.URLParam("electionID")))
+	if err != nil || electionID == nil {
+		return fmt.Errorf("electionID (%q) cannot be decoded", ctx.URLParam("electionID"))
+	}
+	process, err := a.vocapp.State.Process(electionID, true)
+	if err != nil {
+		return fmt.Errorf("cannot get election results: %w", err)
+	}
+	if process == nil {
+		return fmt.Errorf("cannot fetch process %x", electionID)
+	}
+	// do not make distinction between live results elections and encrypted results elections
+	// since we fetch the results from the blockchain state, elections must be terminated and
+	// results must be available
+	if process.Status != models.ProcessStatus_RESULTS {
+		return fmt.Errorf("no results yet results for the election")
+	}
+	// check process results are the same
+	var results [][]*big.Int
+	if len(process.Results) > 0 {
+		firstResult := process.Results[0]
+		for _, processResult := range process.Results[1:] {
+			if processResult == nil {
+				return fmt.Errorf("invalid process result")
+			}
+			// if consensus results do not match, return error
+			for k, questionResult := range processResult.Votes {
+				if questionResult == nil {
+					return fmt.Errorf("invalid question result")
```

Comment:
same as before, we don't want to exit the function here, just continue (also apply to the rest of returns inside the loops)